eb_hooks.py

```diff
@@ -654,6 +655,27 @@ def pre_configure_hook_gromacs(self, *args, **kwargs):
         raise EasyBuildError("GROMACS-specific hook triggered for non-GROMACS easyconfig?!")
 
 
+def pre_configure_hook_llvm(self, *args, **kwargs):
+    """Adjust internal configure options for the LLVM EasyBlock to reinstate filtered out dependencies.
+    In the LLVM EasyBlock, most checks concerning loaded modules are performed at the configure_step.
+    The EB uses a global `general_opts` dict to keep track of options that needs to be reused across stages.
+    The way the EB is structured does allow to inject a CMAKE option through `self._cfgopts` which is a splitted list
+    of the `configure_opts` passed through the EC, but does not allow to override as the `general_opts` dict will
+    take precedence over the `self._cfgopts` list.
+
+    We can instead set the environment variable that EasyBuild uses for `get_software_root` to trick the EB into
+    into pointing to the compat layer.
+    """
+    if self.name == 'LLVM':
```

Comment:
@Crivella Would make sense to also do this for `ROCm-LLVM` (see WIP easyconfigs in https://github.com/bedroge/eb-rocm), so please change to:

```suggestion
    if self.name in ['LLVM', 'ROCm-LLVM']:
```

I'm fine with then with deploying and merging this (especially since we don't have LLVM in EESSI yet)